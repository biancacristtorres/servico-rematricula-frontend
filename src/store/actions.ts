import { ActionTree } from 'vuex';
import { RootState } from '.';
import { AlunoMutationTypes } from './aluno/mutations';
import { StoreNamespaces } from './namespaces';
import { ResumoMutationTypes } from './resumo/mutations';
import { GlobalMutationTypes } from './mutations';
import { GlobalState } from './state';

export enum GlobalActionTypes {
  LIMPAR_STORE = 'LIMPAR_STORE',
  DEFINIR_LINK_PARA_VOLTAR = 'DEFINIR_LINK_PARA_VOLTAR',
  LIMPAR_LINK_PARA_VOLTAR = 'LIMPAR_LINK_PARA_VOLTAR',
}

const actions: ActionTree<GlobalState, RootState> = {

  async [GlobalActionTypes.LIMPAR_STORE]({ commit }) {
    commit(`${ StoreNamespaces.ALUNO }/${ AlunoMutationTypes.LIMPAR_STORE }`);
    commit(`${ StoreNamespaces.RESUMO }/${ ResumoMutationTypes.LIMPAR_STORE }`);
  },

  async [GlobalActionTypes.DEFINIR_LINK_PARA_VOLTAR]({ commit }, link: string) {
    commit(GlobalMutationTypes.SET_LINK_PARA_VOLTAR, link);
  },

  async [GlobalActionTypes.LIMPAR_LINK_PARA_VOLTAR]({ commit }) {
    commit(GlobalMutationTypes.SET_LINK_PARA_VOLTAR, '');
  },
};

export default actions;
