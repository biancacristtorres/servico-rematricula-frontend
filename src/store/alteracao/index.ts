import { Module } from 'vuex';
import { AlteracaoState, state } from './state';
import { RootState } from '@/store';
import mutations from './mutations';
import actions from './actions';
import getters from './getters';

const alteracao: Module<AlteracaoState, RootState> = {
  state,
  mutations,
  actions,
  getters,
  namespaced: true,
};

export default alteracao;
