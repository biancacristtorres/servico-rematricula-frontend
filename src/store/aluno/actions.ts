import { ActionTree } from 'vuex';
import { AlunoState } from './state';
import { RootState } from '@/store';
import { AlunoMutationTypes } from './mutations';
import { Container } from 'typescript-ioc';
import { AlunoService } from '@/services/AlunoService';

export enum AlunoActionTypes {
  OBTER_INFORMACOES_ALUNO = 'OBTER_INFORMACOES_ALUNO',
  LIMPAR_INFORMACOES_ALUNO = 'LIMPAR_INFORMACOES_ALUNO',
  OBTENDO_INFORMACOES_ALUNO = 'OBTENDO_INFORMACOES_ALUNO',
  INFORMACOES_ALUNO_OBTIDAS = 'INFORMACOES_ALUNO_OBTIDAS',
}

const actions: ActionTree<AlunoState, RootState> = {

  async [AlunoActionTypes.OBTER_INFORMACOES_ALUNO]({ commit }, codigoDoAluno: number) {
    const service = (Container.get(AlunoService) as AlunoService);
    const informacoesAluno = await service.obterInformacoesAluno(codigoDoAluno);
    commit(AlunoMutationTypes.SET_INFORMACOES_ALUNO, informacoesAluno);
  },

  async [AlunoActionTypes.LIMPAR_INFORMACOES_ALUNO]({ commit }) {
    commit(AlunoMutationTypes.LIMPAR_STORE);
  },

  async [AlunoActionTypes.OBTENDO_INFORMACOES_ALUNO]({ commit }) {
    commit(AlunoMutationTypes.SET_OBTENDO_INFORMACOES_ALUNO, true);
  },

  async [AlunoActionTypes.INFORMACOES_ALUNO_OBTIDAS]({ commit }) {
    commit(AlunoMutationTypes.SET_OBTENDO_INFORMACOES_ALUNO, false);
  },

};

export default actions;
