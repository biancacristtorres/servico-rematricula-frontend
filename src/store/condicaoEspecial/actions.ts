import { ActionTree } from 'vuex';
import { CondicaoEspecialState } from './state';
import { RootState } from '@/store';
import { MutationsTypes } from './mutations';
import { Container, AutoWired } from 'typescript-ioc';
import { CondicaoEspecialService } from '@/services/CondicaoEspecialService';
import { CondicaoEspecialRequest } from './request/CondicaoEspecialRequest';

export enum ActionTypes {
  PREENCHER_CONDICOES_ESPECIAIS_SOLICITADAS = 'PREENCHER_CONDICOES_ESPECIAIS_SOLICITADAS',
  SOLICITAR_CONDICOES_ESPECIAIS = 'SOLICITAR_CONDICOES_ESPECIAIS',
}

const actions: ActionTree<CondicaoEspecialState, RootState> = {

  async [ActionTypes.PREENCHER_CONDICOES_ESPECIAIS_SOLICITADAS]({ commit, getters }, codigoDoAluno: number) {
    const service = (Container.get(CondicaoEspecialService) as CondicaoEspecialService);
    const solicitadas = await service.solicitadasPor(codigoDoAluno);

    commit(MutationsTypes.SET_SOLICITADAS, solicitadas);
  },

  async [ActionTypes.SOLICITAR_CONDICOES_ESPECIAIS]({ commit }, request: CondicaoEspecialRequest) {
    const service = (Container.get(CondicaoEspecialService) as CondicaoEspecialService);
    await service.solicitar(request.codigoDoAluno, request.condicoesEspeciais);
  },

};

export default actions;
