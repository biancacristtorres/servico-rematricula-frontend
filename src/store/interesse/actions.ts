import { ActionTree } from 'vuex';
import { InteresseState } from './state';
import { RootState } from '@/store';
import { InteresseMutationTypes } from './mutations';
import { Container } from 'typescript-ioc';
import { InteresseService } from '@/services/InteresseService';
import { ManifestarInteresseRequest } from './request/ManifestarInteresseRequest';

export enum InteresseActionTypes {
  OBTER_DISCIPLINAS_PENDENTES = 'OBTER_DISCIPLINAS_PENDENTES',
  MANIFESTAR_INTERESSE = 'MANIFESTAR_INTERESSE',
}

const actions: ActionTree<InteresseState, RootState> = {

  async [InteresseActionTypes.OBTER_DISCIPLINAS_PENDENTES]({ commit, getters }, codigoDoAluno: number) {
    const interesseService = (Container.get(InteresseService) as InteresseService);

    const disciplinas = await interesseService.obterDisciplinasPendentes(codigoDoAluno);

    commit(InteresseMutationTypes.SET_DISCIPLINAS, disciplinas);
  },

  async [InteresseActionTypes.MANIFESTAR_INTERESSE]({ commit }, request: ManifestarInteresseRequest) {
    const interesseService = (Container.get(InteresseService) as InteresseService);

    await interesseService.manifestarInteresse(request.codigoDoAluno, request.disciplinas);
  },

};

export default actions;
