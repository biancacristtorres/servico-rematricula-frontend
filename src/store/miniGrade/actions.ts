import { ActionTree } from 'vuex';
import { MiniGradeState } from './state';
import { RootState } from '@/store';
import { HorarioDaMiniGrade } from '@/model/HorarioDaMiniGrade';
import { MiniGradeMutationTypes } from './mutation';

export enum MiniGradeActionTypes  {
  CARREGAR_MINI_GRADE = 'CARREGAR_MINI_GRADE',
}

const actions: ActionTree<MiniGradeState, RootState> = {

  async [MiniGradeActionTypes.CARREGAR_MINI_GRADE]({ commit }, horarios: HorarioDaMiniGrade[]) {
    commit(MiniGradeMutationTypes.SET_MINI_GRADE, horarios);
  },

};

export default actions;
