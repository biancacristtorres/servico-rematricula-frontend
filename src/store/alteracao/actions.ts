import { ActionTree } from 'vuex';
import { AlteracaoState } from './state';
import { AlteracaoMutationsTypes } from './mutations';
import store, { RootState, StoreNamespaces } from '@/store';
import { Container } from 'typescript-ioc';
import { DisciplinaDisponivelService } from '@/services/alteracao/DisciplinaDisponivelService';
import { TurmasActionRequest } from './request/TurmasActionRequest';
import { TurmasService } from '@/services/alteracao/TurmasService';
import { Turma } from '@/model/Turma';
import { TurmasMutationRequest } from './request/TurmasMutationRequest';
import { TurmasEquivalentesService } from '@/services/alteracao/TurmasEquivalentesService';
import { TurmasEquivalentesMutationRequest } from './request/TurmasEquivalentesMutationRequest';
import { AlternarTurmaAtualActionRequest } from './request/AlternarTurmaAtualActionRequest';
import { AlternarTurmaAtualRequest } from './request/AlternarTurmaAtualRequest';
import { TurmasEspeciaisService } from '@/services/alteracao/TurmasEspeciaisService';
import { TurmasEspeciaisMutationRequest } from './request/TurmasEspeciaisMutationRequest';
import { ConcluirAlteracaoDeSucessoActionRequest } from './request/ConcluirAlteracaoDeSucessoActionRequest';
import { Filtro } from '@/model/Filtro';
import { MiniGradeMutationTypes } from '../miniGrade/mutation';

export enum AlteracaoActionTypes {
  DISCIPLINAS = 'DISCIPLINAS',
  TURMAS = 'TURMAS',
  TURMAS_EQUIVALENTES = 'TURMAS_EQUIVALENTES',
  ALTERNAR_TURMA_ATUAL = 'ALTERNAR_TURMA_ATUAL',
  ALTERANDO_TURMAS = 'ALTERANDO_TURMAS',
  ALTERACAO_FINALIZADA = 'ALTERACAO_FINALIZADA',
  PREPARANDO_TURMAS = 'PREPARANDO_TURMAS',
  PREPARACAO_FINALIZADA = 'PREPARACAO_FINALIZADA',
  TURMAS_ESPECIAIS = 'TURMAS_ESPECIAIS',
  INFORMAR_ALTERACAO = 'INFORMAR_ALTERACAO',
  LIMPAR_ALTERACAO = 'LIMPAR_ALTERACAO',
  CONCLUIR_ALTERACAO_DE_SUCESSO = 'CONCLUIR_ALTERACAO_DE_SUCESSO',
  ALTERAR_FILTRO = 'ALTERAR_FILTRO',
}

const actions: ActionTree<AlteracaoState, RootState> = {

  async [AlteracaoActionTypes.DISCIPLINAS]({ commit }, codigoDoAluno: number) {

    const service = (Container.get(DisciplinaDisponivelService) as DisciplinaDisponivelService);

    const disciplinas = await service.obter(codigoDoAluno);

    commit(AlteracaoMutationsTypes.SET_DISCIPLINAS, disciplinas);
  },

  async [AlteracaoActionTypes.TURMAS]({ commit }, actionRequest: TurmasActionRequest) {
    const service = (Container.get(TurmasService) as TurmasService);

    const turmas: Turma[] = await service.obterPor(
      actionRequest.codigoDaDisciplina, actionRequest.codigoDoAluno,
    );

    const mutationRequest = new TurmasMutationRequest(
      actionRequest.codigoDaDisciplina, turmas,
    );

    commit(AlteracaoMutationsTypes.SET_TURMAS, mutationRequest);
  },

  async [AlteracaoActionTypes.TURMAS_EQUIVALENTES]({ commit }, actionRequest: TurmasActionRequest) {
    const service = (Container.get(TurmasEquivalentesService) as TurmasEquivalentesService);

    const equivalentes: Turma[] = await service.obterPor(
      actionRequest.codigoDaDisciplina, actionRequest.codigoDoAluno,
    );

    const mutationRequest = new TurmasEquivalentesMutationRequest(
      actionRequest.codigoDaDisciplina, equivalentes,
    );

    commit(AlteracaoMutationsTypes.SET_TURMAS_EQUIVALENTES, mutationRequest);
  },

  async [AlteracaoActionTypes.TURMAS_ESPECIAIS]({ commit }, actionRequest: TurmasActionRequest) {
    const service = (Container.get(TurmasEspeciaisService) as TurmasEspeciaisService);

    const especiais: Turma[] = await service.obterPor(
      actionRequest.codigoDaDisciplina, actionRequest.codigoDoAluno,
    );

    const mutationRequest = new TurmasEspeciaisMutationRequest(
      actionRequest.codigoDaDisciplina, especiais,
    );

    commit(AlteracaoMutationsTypes.SET_TURMAS_ESPECIAIS, mutationRequest);
  },

  async [AlteracaoActionTypes.ALTERNAR_TURMA_ATUAL]({ commit }, actionRequest: AlternarTurmaAtualActionRequest) {
    const mutationRequest = new AlternarTurmaAtualRequest(
      actionRequest.codigoDaDisciplina, actionRequest.codigoUnico,
    );

    commit(AlteracaoMutationsTypes.ALTERNAR_TURMA_ATUAL, mutationRequest);
  },

  async [AlteracaoActionTypes.ALTERANDO_TURMAS]({ commit }) {
    commit(AlteracaoMutationsTypes.SET_ALTERANDO, true);
  },

  async [AlteracaoActionTypes.ALTERACAO_FINALIZADA]({ commit }) {
    commit(AlteracaoMutationsTypes.SET_ALTERANDO, false);
  },

  async [AlteracaoActionTypes.PREPARANDO_TURMAS]({ commit }) {
    commit(AlteracaoMutationsTypes.SET_PREPARANDO_TURMAS, true);
  },

  async [AlteracaoActionTypes.PREPARACAO_FINALIZADA]({ commit }) {
    commit(AlteracaoMutationsTypes.SET_PREPARANDO_TURMAS, false);
  },

  async [AlteracaoActionTypes.INFORMAR_ALTERACAO]({ commit }) {
    commit(AlteracaoMutationsTypes.SET_ALTERACAO);
  },

  async [AlteracaoActionTypes.LIMPAR_ALTERACAO]({ commit }) {
    commit(AlteracaoMutationsTypes.LIMPAR_ALTERACAO);
  },

  async [AlteracaoActionTypes.CONCLUIR_ALTERACAO_DE_SUCESSO](
    { commit },
    actionRequest: ConcluirAlteracaoDeSucessoActionRequest) {
      const mutationRequest = new AlternarTurmaAtualRequest(
        actionRequest.codigoDaDisciplina, actionRequest.codigoUnico, actionRequest.eTurmaAtual,
      );

      commit(AlteracaoMutationsTypes.ALTERNAR_TURMA_ATUAL, mutationRequest);
      commit(AlteracaoMutationsTypes.SET_ALTERACAO);

      store.commit(`${ StoreNamespaces.MINI_GRADE }/${ MiniGradeMutationTypes.SET_MINI_GRADE }`,
        actionRequest.horariosDaMiniGrade);
  },

  async [AlteracaoActionTypes.ALTERAR_FILTRO]({ commit }, novoFiltro: Filtro) {
    commit(AlteracaoMutationsTypes.SET_NOVO_FILTRO, Object.assign({}, novoFiltro));
  },
};

export default actions;
