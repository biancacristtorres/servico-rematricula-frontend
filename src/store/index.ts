import Vue from 'vue';
import Vuex from 'vuex';
import autorizacao from 'componente-frontend-core/store/autorizacao';
import interesse from './interesse';
import condicaoEspecial from './condicaoEspecial';
import featureToggle from 'componente-frontend-core/store/featureToggle';
import alteracao from './alteracao';
import resumo from './resumo';
import aluno from './aluno';
import actions from './actions';
import marca from './marca';
import miniGrade from './miniGrade';
import { state } from './state';
import mutations from './mutations';

export { StoreNamespaces } from '@/store/namespaces';

Vue.use(Vuex);

export class RootState {
}

const modules = {
  autorizacao,
  interesse,
  condicaoEspecial,
  featureToggle,
  alteracao,
  resumo,
  aluno,
  marca,
  miniGrade,
};

const store = new Vuex.Store({
  modules,
  actions,
  state,
  mutations,
});

Object.assign(window, { $store: store });

export default store;
