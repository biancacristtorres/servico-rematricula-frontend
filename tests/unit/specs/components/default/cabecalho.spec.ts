
import { CabecalhoFactory } from '../../../factory/components/CabecalhoFactory';
import store, { StoreNamespaces } from '@/store';
import { AlteracaoActionTypes } from '@/store/alteracao/actions';

describe('Componente de Cabeçalho', () => {
  const cabecalhoFactory = new CabecalhoFactory();

  afterEach(() => {
    cabecalhoFactory.destruirDialogo();
  });

  it('O título exibido deve ser o título passado por parâmetro', async (done) => {
    cabecalhoFactory.dadoOTitulo('XPTO');

    await cabecalhoFactory.montarComponente();

    expect(cabecalhoFactory.tituloDoCabecalho).toEqual('XPTO');
    done();
  });

  it('Quando houver link para voltar, então um botão de voltar deve ser exibido', async (done) => {
    await cabecalhoFactory.montarComponente();

    cabecalhoFactory.dadoOLinkParaVoltar('/Resumo');

    expect(cabecalhoFactory.linkDoBotaoVoltar).toEqual('/Resumo');
    expect(cabecalhoFactory.existeOBotaoVoltar).toBeTruthy();
    done();
  });

  it('Quando não houver link para voltar, então o botão de voltar não deve ser exibido ', async (done) => {
    cabecalhoFactory.dadoQueNaoExisteLinkParaVoltar();

    await cabecalhoFactory.montarComponente();

    expect(cabecalhoFactory.existeOBotaoVoltar).toBeFalsy();
    done();
  });

  it('Quando a rota for de erro de sistema, ajuda e protocolos e pessoa logada não são exibidos', async (done) => {
    cabecalhoFactory.dadoQueEstouNaTelaDeErroDeSistema();

    await cabecalhoFactory.montarComponente();

    const ajudaEProtocolos = cabecalhoFactory.componente.find('#ajuda-e-protocolos');
    const pessoaLogada = cabecalhoFactory.componente.find('#pessoa-logada');

    expect(ajudaEProtocolos.exists()).toBeFalsy();
    expect(pessoaLogada.exists()).toBeFalsy();
    done();
  });

  it('Quando existe alteração, um feedback é exibido ao voltar', async (done) => {
    store.dispatch(`${ StoreNamespaces.ALTERACAO }/${ AlteracaoActionTypes.INFORMAR_ALTERACAO }`);

    await cabecalhoFactory.montarComponente();
    cabecalhoFactory.dadoOLinkParaVoltar('/Resumo');
    cabecalhoFactory.componente.find('#link-para-voltar').trigger('click');

    expect(cabecalhoFactory.oDialogoEstaAberto()).toBeTruthy();
    done();
  });

  it('Quando não existe alteração, nenhum feedback é exibido ao voltar', async (done) => {
    store.dispatch(`${ StoreNamespaces.ALTERACAO }/${ AlteracaoActionTypes.LIMPAR_ALTERACAO }`);

    await cabecalhoFactory.montarComponente();
    cabecalhoFactory.dadoOLinkParaVoltar('/Resumo');
    cabecalhoFactory.componente.find('#link-para-voltar').trigger('click');

    expect(cabecalhoFactory.oDialogoEstaAberto()).toBeFalsy();
    done();
  });

  it('Snapshot', async (done) => {
    await cabecalhoFactory.montarComponente();

    expect(cabecalhoFactory.componente).toMatchSnapshot();
    done();
  });
});
