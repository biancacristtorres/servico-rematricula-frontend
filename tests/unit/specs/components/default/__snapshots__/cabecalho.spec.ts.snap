// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Componente de Cabe√ßalho Snapshot 1`] = `
<vtoolbar-stub scrollthreshold="300" color="primary" dark="">
  <!---->
  <vtoolbartitle-stub id="titulo-cabecalho">
    XPTO
  </vtoolbartitle-stub>
  <vspacer-stub></vspacer-stub>
  <!---->
  <!---->
</vtoolbar-stub>
`;
